crates/oxc_ast/src/trivia.rs

```diff
@@ -57,6 +57,17 @@ impl Trivias {
         self.comments.range(span.start..span.end).count() > 0
     }
 
+    #[must_use]
+    pub fn get_leading_comment_span(&self, span: Span) -> Option<Span> {
+        let comments = self.comments.range(..span.start);
+        match comments.last() {
+            Some((start, comment)) if comment.end() + 2 == span.start => {
+                Some(Span::new(*start, comment.end()))
+            }
+            _ => None,
+        }
+    }
```

Comment:
This is wrong but I'm not entirely sure how we can determine that a comment span is attached to a particular node span